### PR TITLE
jwt claim based routing promotion

### PR DESCRIPTION
**Pull request by aryan16**
Promotion of jwt claim based routing from experimental to alpha

### CODE REVIEW

**Comment by ericvn** [comment]
I expect there should be an appropriate features/... md file as well with the appropriate checkmarks.

I did look at the open PRs where @johnma14 created a number of them for the existing features.

**Comment by aryan16** [comment]
> I expect there should be an appropriate features/... md file as well with the appropriate checkmarks.
> 
> I did look at the open PRs where @johnma14 created a number of them for the existing features to see if there was something done already. But since this looks to be a newly documented feature, you will have to create one. I don't see a template md to copy from.

Do we really want a separate feature.md for this ? It's a feature which is dependent on reqAuth and Virtual Service and we have a feature.md for both of them. We do have many features where feature.ms is missing - https://github.com/istio/enhancements/blob/master/features.yaml#L62

**Comment by aryan16** [comment]
@ericvn @howardjohn PTAL

File: features/claim_based_routing.md [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+- [ ] Any necessary tooling to use/debug the feature has been implemented and is complete. 
+
+**Bugs**
+
+- [] Feature has no known major issues.
+
+**Approvals**
+
+- [] The appropriate work group(s) have reviewed and approved promotion of the feature.
```
Comment (linsun) [inline features/claim_based_routing.md:197]: Can you get this checked?

**Review by linsun — APPROVED** [review]

File: features.yaml [hunk 2]
```diff
@@ -251,6 +251,14 @@ features:
       maturity: Alpha
       nextExpectedPromotion: "1.14"
     area: Security and policy enforcement
+  - name: "JWT Claim Based Routing"
```
Comment (linsun) [inline features.yaml:254]: can we highlight in the name this feature is only for ingress?

File: features/claim_based_routing.md [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+[//]: # (The feature implementation section must be completed before submission of the document.)
+
+# Feature:
+
+[//]: # (All information in this section is mandatory.)
+
+**Feature name:**
+
+JWT Claim Based Routing
```
Comment (ericvn) [inline features/claim_based_routing.md:17]: I can't do suggestions (maybe as this is merged), and I think the TOC might want to provide some guidance on how to accurately fill these form out. So of the next time we want to update this document, based on some early docs that Brian did, I think the lines in the header can be on a single line, so something like:

**Feature name:**  JWT Claim Based Routing

for this and following header sections.

File: features/claim_based_routing.md [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+
+> The task page of the feature is published at https://istio.io/latest/docs/tasks/security/authentication/jwt-route/
+
+- [x] Basic feature docs are published on preliminary.istio.io describing what the feature does, how to use it, and any caveats. 
+
+> A task has been added
+
+- [x] Release notes entries added as appropriate
+
+> Targeting for 1.16 release, will add it later when required
```
Comment (ericvn) [inline features/claim_based_routing.md:122]: You could point to the PR that has the RN.

File: features/claim_based_routing.md [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+> The alpha promotion of the feature is fully compatible with the experimental version, we do not need upgrade notes.
+
+**Tests**
+
+- [x] Automated integration tests cover core use cases with the feature enabled. 
+- [x] When disabled, the feature does not affect system stability or performance. 
+
+**API**
+
+- [ ] Initial API review.
```
Comment (ericvn) [inline features/claim_based_routing.md:135]: Is there a PR or maybe this is NA?

**Review by ericvn — COMMENTED** [review]
Approved - I did make a few comments for changes when the file is updated for Beta. They are nits. I think the TOC should have another filled out Template to show how we would be reviewing these in the future to better guide the authors. Even I have had my thoughts change as I have reviewed a few PRs.

File: features/claim_based_routing.md (hunk 4 above)
Comment (aryan16) [reply features/claim_based_routing.md:122]: sure, will add it once I add 1.16 release notes.

File: features/claim_based_routing.md (hunk 5 above)
Comment (aryan16) [reply features/claim_based_routing.md:135]: It's NA. There's no impact on API

File: features.yaml (hunk 2 above)
Comment (aryan16) [reply features.yaml:254]: sure, will do it once I add a PR for release note in the same file.

**Comment by therealmitchconnors** [comment]
LGTM

**Comment by nrjpoddar** [comment]
We should fill the experimental section as I assume we have addressed those requirements as well.

**Comment by louiscryan** [comment]
LGTM

**Pull request merged**